### PR TITLE
[16.0][ADD] pos_membership_extension

### PR DESCRIPTION
**Pull request by legalsylvain**
Note : CI is red because of unmerged dependencies : see : https://github.com/OCA/vertical-association/pull/132)

### Description

This module extends the functionality of point of sale module to implement features related to ``membership_extension`` OCA module. (https://github.com/OCA/vertical-association repository) For the time being, it prevents cashier to sell product to customer that doesn't belong to some membership categories.

### Configure

* open a product form
* Go to the sale tab
* fill the field "Allowed Membership Categories"

product_form

* Configure also the membership categories. For that purpose, refer to the documentation of the OCA ``membership_extension`` module.

### Usage

* open the Point of Sale product

* You can see that some products have a new icon that mention that the product is not sallable. the icon disappear if you select a customer that belong to one of the membership categories :

point_of_sale_product_item

* The module adds in the Product Pop Up info, the list of the membership categories allowed for the current product :

point_of_sale_popup_product_info

* If you select the product with an incorrect customer, you'll see an error Popup :

point_of_sale_popup_error

* If you selected allowed restricted product for a partner, and then switch for another partner, a check is realized to verify that the selected products are allowed. If not, order lines will be removed and a popup message will be displayed :

point_of_sale_switch_partner_popup

### CODE REVIEW

**Comment by legalsylvain** [comment]
CC : @PierrickBrun , could you review ?

File: pos_membership_extension/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2019 Onestein - Andrea Stirpe
```
Comment (PierrickBrun) [inline pos_membership_extension/models/product_product.py]: I think this came from a copied file

File: pos_membership_extension/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2019 Onestein - Andrea Stirpe
```
Comment (PierrickBrun) [inline pos_membership_extension/models/product_template.py]: same as before

**Review by PierrickBrun — APPROVED** [review]
Except the 2 comments everything looks perfect.

Can you rebase since the PR it depends on is merged ?

File: pos_membership_extension/models/product_template.py (hunk 2 above)
Comment (legalsylvain) [reply pos_membership_extension/models/product_template.py]: Done.

**Review by PaulGoubert — APPROVED** [review]

**Comment by legalsylvain** [comment]
thanks for your reviews.

/ocabot merge patch

**Pull request merged**